gpttrace/bpftrace.py

```diff
@@ -0,0 +1,295 @@
+                    "type": "string",
+                    "description": "output buffering mode"
+                },
+                "format": {
+                    "type": "string",
+                    "description": "output format"
+                },
+                "outputFile": {
+                    "type": "string",
+                    "description": "redirect bpftrace output to file"
+                },
+                "debugInfo": {
+                    "type": "boolean",
+                    "description": "debug info dry run"
+                },
+                "verboseDebugInfo": {
+                    "type": "boolean",
+                    "description": "verbose debug info dry run"
+                },
+                "program": {
+                    "type": "string",
+                    "description": "program to execute"
+                },
+                "includeDir": {
+                    "type": "array",
+                    "items": {
+                        "type": "string"
+                    },
+                    "description": "directories to add to the include search path"
+                },
+                "usdtFileActivation": {
+                    "type": "boolean",
+                    "description": "activate usdt semaphores based on file path"
+                },
+                "unsafe": {
+                    "type": "boolean",
+                    "description": "allow unsafe builtin functions"
+                },
+                "quiet": {
+                    "type": "boolean",
+                    "description": "keep messages quiet"
+                },
+                "verbose": {
+                    "type": "boolean",
+                    "description": "verbose messages"
+                },
+                "noWarnings": {
+                    "type": "boolean",
+                    "description": "disable all warning messages"
+                },
+                "timeout": {
+                    "type": "integer",
+                    "description": "seconds to run the command"
+                },
+                "continue": {
+                    "type": "boolean",
+                    "description": "finish conversation and not continue."
+                }
+            },
+            "required": ["program"]
+        }
+    },
+    {
+        "name": "SaveFile",
+        "description": "Save the eBPF program to file",
+        "parameters": {
+            "type": "object",
+            "properties": {
+                "filename": {
+                    "type": "string",
+                    "description": "the file name to save to"
+                },
+                "content": {
+                    "type": "string",
+                    "description": "the file content"
+                }
+            },
+            "required": ["filename", "continue"]
+        }
+    }
+]
+
+
+def run_command_with_timeout(command: str, timeout: int) -> dict:
```

Comment:
Consider TypedDict if the return value is a dict with fixed keys and value types

Reply:
Thanks! it's added.